File: Ejercicios_POO3.1_Java/src/Ejercicio3_JuegoRol/Personaje.java
package Ejercicio3_JuegoRol;

public class Personaje {
    protected String nombre;
    public int puntosVida;

    public Personaje(String nombre, int puntosVida) {
        this.nombre = nombre;
        this.puntosVida = puntosVida;
    }

    public int atacar() {
        return 0;
    }

    public void usarHechizo(Hechizo hechizo) {
        System.out.println(nombre + " no puede usar hechizos.");
    }

    public boolean estaVivo() {
        return puntosVida > 0;
    }

    @Override
    public String toString() {
        return nombre + " (Vida: " + puntosVida + ")";
    }
}

File: Ejercicios_POO3.1_Kotlin/src/Ejercicio3_JuegoRol/Personaje.kt
package Ejercicio3_JuegoRol


/*
 la clase personaje es open para que pueda ser heredada por otras clases
 esto significa que se puede sobreescribir sus métodos y propiedades

 */
open class Personaje(val nombre: String, var puntosVida: Int) {
    open fun atacar(): Int {
        return 0 // Método que debe ser sobreescrito
    }

    open fun usarHechizo(hechizo: Hechizo) {
        println("$nombre no puede usar hechizos.")
    }

    fun estaVivo(): Boolean {
        return puntosVida > 0
    }

    override fun toString(): String {
        return "$nombre (Vida: $puntosVida)"
    }
}